src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DoNotCallEnumerableCastThatWillFailAnalyzerTests.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+using Microsoft.CodeAnalysis.Testing;
+
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DoNotCallEnumerableCastThatWillFailAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+using VerifyVB = Test.Utilities.VisualBasicCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Runtime.DoNotCallEnumerableCastThatWillFailAnalyzer,
+    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class DoNotCallEnumerableCastThatWillFailAnalyzerTests
+    {
+        [Fact]
+        public async Task DiagnosticCasesCSharp()
+        {
+            var ofTypeRule = DoNotCallEnumerableCastThatWillFailAnalyzer.OfTypeRule;
+
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System.Linq;
+
+class C
+{
+    public void M()
+    {
+        var a = (new int[0]).OfType<object>();
+        var b = (new int[0]).OfType<string>();
+        var c = (new object[0]).OfType<string>();
+    }
+}
+",
+    // /0/Test0.cs(9,17): info CA9999: If the source sequence contains any elements, the sequence returned by Cast will throw InvalidCastException at runtime when enumerated.
+    VerifyCS.Diagnostic(ofTypeRule).WithSpan(9, 17, 9, 46).WithArguments("int", "string")
```

Comment:
Could you please use inline markup notation rather than `WithSpan` it is easier to read and comprehend (e.g. `var a = (new int[0]).{|#0:OfType<object>()|}`

Reply:
This is the format output by the test runner - is there any way to get it in this format or will this have to be done manually?

Reply:
Will do.